src/main/java/net/clgd/ccemux/Utils.java

```diff
@@ -0,0 +1,79 @@
+package net.clgd.ccemux;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Random;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.MoreExecutors;
+import dan200.computercraft.api.lua.ILuaContext;
+import dan200.computercraft.api.lua.LuaException;
+import net.clgd.ccemux.api.emulation.EmulatedComputer;
+
+public final class Utils {
+	private static final int RETRIES = 10;
+	private static Random random = new Random();
+
+	private Utils() {
+	}
+
+	public static synchronized File createUniqueFile(File directory, String name, String extension) throws IOException {
```

Comment:
Would `Files.createTempFile` work here?

Reply:
It _always_ adds a unique suffix, which I was trying to avoid here. I could probably replace the second half of the function of this though.

Reply:
It has an overload that accepts both a prefix and a suffix, only adding random junk to the middle of the name